caluma/caluma_form/models.py

```diff
@@ -295,11 +295,13 @@ def copy(self, family=None, user=None):
 
         return new_document
 
+    def is_root(self):
+        return self == self.family
+
     @cached_property
     def last_modified_answer(self):
-        return (
-            Answer.objects.filter(document__family=self).order_by("modified_at").last()
-        )
+        _filter = "document__family" if self.is_root() else "document"
+        return Answer.objects.filter(**{_filter: self}).order_by("-modified_at").first()
```

Comment:
Is it really necessary to differentiate? Can't we just do `Answer.objects.filter(Q(document=self) | Q(document__family=self)).order_by("modified_at").last()`?

Reply:
That should also work and is much nicer to read! I think we could save a join by differentiating but it's not worth it.